File: src/main/java/com/server/eFacture/Entity/Devis/Enregistrement.java
package com.server.eFacture.Entity.Devis;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.server.eFacture.Entity.Entreprise.Materiel;
import com.server.eFacture.Entity.Entreprise.Tache;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table
@Data
public class Enregistrement {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id ;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
    private Devis devis;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
    private Tache tache;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
    private Materiel materiel;

    private Integer quantite;
}

File: src/app/Core/Model/Devis/Enregistrement.ts
import { Materiel } from "../Entreprise/Materiel";
import { Devis } from "./Devis";
import { Tache } from "./Tache";

export interface Enregistrement{
    id:number; 
    devis :Devis; 
    tache :Tache ; 
    materiel : Materiel ; 
    quantite : number ; 
}